tests/mocks/MockOperationStore.ts

```diff
@@ -73,6 +73,9 @@ export default class MockOperationStore implements OperationStore {
     // Sort needed if there was a put operation since last sort.
     if (updatedSinceLastSort) {
       didOps.sort(compareOperation);       // in-place sort
+      didOps = didOps.filter((elem, index, self) => {  // remove duplicates
+        return (index === 0) || compareOperation(elem, self[index - 1]) !== 0;
```

Comment:
This looks a bit like hack that is designed to handle the above tests, why not just prevent duplicate in the `putBatch()`?

Reply:
I am not convinced this is a hack: (1) since we are anyway sorting before returning from get(), it is efficient to use the sorted operations to deduplicate; (2) the result of deduplication is "materialized" from this point on and the next get() operation would not do the deduplication. 

If we push this logic to putBatch() we would have to push the sorting as well (or use an inefficient method) - but we would be doing more work in the process(batch) -> putBatch() code path and our design with deferred processing is to make this code path very inexpensive.

Does this make sense?

Reply:
Sort of. I'm not going to be too picky on this since this is a mock, but my point was that if this was a real store, then I think we wouldn't have stored duplicates, so the comment was trying to align with what a real store would behave.